wp-includes/components/class-wp-service-worker-caching-routes.php

```diff
@@ -137,66 +242,48 @@ public function get_all() {
 		return $this->routes;
 	}
 
-
 	/**
 	 * Prepare caching strategy args for export to JS.
 	 *
 	 * @since 0.2
 	 *
-	 * @param array $strategy_args Strategy args.
+	 * @param array $strategy_args Strategy args. Already normalized with snake_case keys.
 	 * @return string JS IIFE which returns object for passing to registerRoute.
 	 */
 	public static function prepare_strategy_args_for_js_export( $strategy_args ) {
 		$exported = '( function() {';
 
-		// Extract plugins since not JSON-serializable as-is.
+		// Pluck out plugins defined at the top-level.
 		$plugins = array();
-		if ( isset( $strategy_args['plugins'] ) ) {
-			$plugins = $strategy_args['plugins'];
-			unset( $strategy_args['plugins'] );
-		}
-
-		foreach ( $strategy_args as $strategy_arg_name => $strategy_arg_value ) {
-			if ( false !== strpos( $strategy_arg_name, '_' ) ) {
-				$strategy_arg_name = preg_replace_callback( '/_[a-z]/', array( __CLASS__, 'convert_snake_case_to_camel_case_callback' ), $strategy_arg_name );
+		foreach ( self::WORKBOX_CORE_PLUGINS as $plugin_name ) {
+			if ( array_key_exists( $plugin_name, $strategy_args ) ) {
```

Comment:
Mixing plugin args with other args feels a bit error-prone to me - maybe we can keep the plugin-specific data in a `plugins` or `plugin_config` sub-key associative array like it was before? I think this will better prepare it for a potential future where we also allow customizability via other plugins than those we "recognize".

Something like:
```suggestion
			if ( ! empty( $strategy_args['plugins'] ) && array_key_exists( $plugin_name, $strategy_args['plugins'] ) ) {
```

As far as I can tell this will also keep backward-compatibility in the format which would be a nice side-effect?

Reply:
I actually inquired about this with Jeff Posnick. I was also hesitant about whether flattening the config would cause conflicts. He said it would not be a problem, and in fact the flattened config is similar to what Workbox.js calls a [`RuntimeCachingEntry`](https://developers.google.com/web/tools/workbox/reference-docs/latest/module-workbox-build#.RuntimeCachingEntry) which you can see defines the plugin-specific configuration alongside strategy-specific configuration. It does have an additional `plugins` key:

![image](https://user-images.githubusercontent.com/134745/99158763-35ceff80-268b-11eb-9173-e68ae95e2fb9.png)

But this isn't relevant in the PHP API since all of the plugins are among a fixed set:

https://github.com/GoogleChromeLabs/pwa-wp/blob/f1381dfe593dfc4df281bc82720bddb529db2df3/wp-includes/components/class-wp-service-worker-caching-routes.php#L61-L67

We limit the set because it's how we know how to construct the plugin object:

https://github.com/GoogleChromeLabs/pwa-wp/blob/f1381dfe593dfc4df281bc82720bddb529db2df3/wp-includes/components/class-wp-service-worker-caching-routes.php#L280-L285

I think it's advantageous to remove the `plugins` key from the configuration because it can be confusing for WP developers who have a different conception of what a plugin is. 

If someone wants to create a custom plugin, they can do so but they'd have to register it via JS instead of the PHP API.

Reply:
Okay that makes sense then. Thanks for clarifying.